jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ConcurrentPool.java

```diff
@@ -319,6 +340,19 @@ public void dump(Appendable out, String indent) throws IOException
             new DumpableCollection("entries", entries));
     }
 
+    public int sweep()
+    {
+        List<Entry<P>> toRemove = new ArrayList<>();
+        for (Entry<P> entry : entries)
+        {
+            ConcurrentEntry<P> ce = (ConcurrentEntry<P>)entry;
+            if (ce.isGarbage())
+                toRemove.add(entry);
```

Comment:
I think we should log a warning if we ever see a collected acquired buffer.   We can have an option to not warn, but I think by default we should warn as we really do not want to encourage leaking as a way forward.

Reply:
Agreed, we should warn on detected leaks.